plugins/modules/ssh_config.py

```diff
@@ -274,6 +283,12 @@ def ensure_state(self):
             controlpersist=fix_bool_str(self.params.get('controlpersist')),
             dynamicforward=self.params.get('dynamicforward'),
         )
+        if self.params.get('other_options'):
+            for key, value in self.params.get('other_options').items():
+                if key not in args:
+                    if not isinstance(value, string_types):
+                        self.module.fail_json(msg="The other_options value provided for key %s must be a string" % key)
```

Comment:
How about:
```suggestion
                        self.module.fail_json(msg="The other_options value provided for key {key!r} must be a string".format(key=key))
```
That way the key is put in quotes automatically.

Or even better, let's include `value`'s type:
```suggestion
                        self.module.fail_json(msg="The other_options value provided for key {key!r} must be a string, got {type}".format(key=key, type=type(value)))
```
That might give the user a hint what's going wrong. (Ansible tends to do some type conversions behind the user's back, sometimes in a good way, but sometimes also in a bad way.)